packages/@angular/cli/tasks/npm-install.ts

```diff
@@ -11,7 +12,7 @@ export default Task.extend({
       packageManager = 'npm';
     }
 
-    return new Promise(function(resolve, reject) {
+    return checkYarnOrCNPM().then(function () {
```

Comment:
If you mean why I did not write `return checkYarnOrCNPM().then(() => {` then the answer will be just because I did not see any arrow functions in code I have touched so decided not to for consistency